frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/storage-tab-state-update.ts

```diff
@@ -200,9 +201,46 @@ export const internalStorageDiskBusUpdater = ({
   }
 };
 
+const initialStorageClassUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+
+  if (
+    !hasVMSettingsValueChanged(prevState, state, id, VMSettingsField.DEFAULT_STORAGE_CLASS) &&
+    !hasStoragesChanged(prevState, state, id)
+  ) {
+    return;
+  }
+  const storageClassName = iGetVmSettingValue(state, id, VMSettingsField.DEFAULT_STORAGE_CLASS);
+
+  if (storageClassName) {
+    const iProvisionSourceStorage = iGetProvisionSourceStorage(state, id);
+    const provisionSourceStorage: VMWizardStorage =
+      iProvisionSourceStorage && iProvisionSourceStorage.toJSON();
+
+    if (
+      provisionSourceStorage &&
+      !getDataVolumeStorageClassName(provisionSourceStorage?.dataVolume)
```

Comment:
this might be too forceful. What about the storages with no SC?

Reply:
@suomiy 
this basically means update the sc to the default one if the storage doesn't have any sc, whats wrong?
if we'll remove this line the sc will always reset to the default one even if the user chooses a different sc.

Reply:
There might be also an option of not having the storage class for creating the templates too. Since the default SC might change when the template gets used.

Reply:
edit: ah sorry for the first point - in  that case there will not be any SC to select.

So what is left is to allow selecting empty SC in disk modal

Reply:
https://github.com/openshift/console/pull/7715